fabtests/unit/mr_test.c

```diff
@@ -188,16 +206,73 @@ static int mr_regattr()
 	return TEST_RET_VAL(ret, testret);
 }
 
+static int mr_reg_free_then_alloc()
+{
+	int i, ret, testret;
+	size_t buf_size = test_size[test_cnt-1].size;
+	struct fid_mr *mr;
+	char *buf2;
+	struct iovec iov;
+	struct fi_mr_attr attr = {0};
+
+	iov.iov_base = buf;
+	iov.iov_len = buf_size;
+
+	attr.mr_iov = &iov;
+	attr.iov_count = 1;
+	attr.access = FI_SEND;
+	attr.offset = 0;
+	attr.requested_key = FT_MR_KEY;
+	attr.context = NULL;
+	attr.iface = opts.iface;
+	if (opts.iface == FI_HMEM_CUDA)
+		attr.device.cuda = opts.device;
+
+	testret = FAIL;
+	for (i=0; i < 5; ++i) {
```

Comment:
5 is a magic number? Could it be a named constant instead?

Reply:
added a variable `numtry`